doc/decisions/0_drafts/keyset_types.md

```diff
@@ -0,0 +1,79 @@
+# Types of `KeySet`s
+
+## Problem
+
+A `KeySet` can be used in different ways.
+Among these are at least:
+
+- As originally intended, a set of configuration data
+- Metadata of a `Key`
+- General associative array data structure
+
+These types have slightly different requirements, and therefore may need to be treated differently.
+
+## Constraints
+
+- API duplication just to separate types should be avoided
+- API must be safe to use for each of the different types
+
+## Assumptions
+
+- The types are not different enough to warrant the overhead of entirely different APIs and data structures.
+
+## Considered Alternatives
+
+### No restrictions
+
+Leave `KeySet` entirely unrestricted, so it can be used for each use case.
+
+Sometimes there can be odd edge cases.
+For example, what should a storage plugin do, if it receives a `KeySet` without both configuration data and metadata.
```

Comment:
Put this up to the problem.

Reply:
Just noticed a typo here:

```suggestion
Sometimes there can be odd edge cases.
For example, what should a storage plugin do, if it receives a `KeySet` with both configuration data and metadata.
```

I'll add it to the problem as well.